src/net_processing.cpp

```diff
@@ -1605,7 +1622,21 @@ void static ProcessGetData(CNode* pfrom, const CChainParams& chainparams, CConnm
                     push = true;
                 }
             }
-            if (!push) {
+
+            if (push) {
+                int num = 0;
+
+                // We intepret processing a GETDATA for a transaction as a successful initial broadcast
+                // and remove it from our unbroadcast set.
+                {
+                    LOCK(mempool.cs);
+                    num = mempool.m_unbroadcast_txids.erase(inv.hash);
```

Comment:
Needs rebase to make the `const CTxMemPool&` non-const

Reply:
thanks!